tools/azsdk-cli/docs/specs/0-scenario-2.spec.md

```diff
@@ -0,0 +1,957 @@
+
+- Clear success benchmarks (including scaffolding)
+- Repeatable validation across customization and testing
+- Explicit scope for TypeSpec vs. code customization and supporting **[Test Infrastructure](#test-infrastructure)**
+- Cross-language consistency for new stages
+- Confirmation that Scenario 1 capabilities integrate cleanly
+
+---
+
+## Context & Assumptions
+
+### Environment
+
+- Windows, macOS, or Linux machine with freshly cloned repositories (`azure-rest-api-specs` plus all five language repositories)
+- TypeSpec modifications are **local only**
+- Agent-mode interactions occur in **VS Code, Visual Studio, or IntelliJ with GitHub Copilot** with `azure-rest-api-specs` open
+- Azure subscription permits on-demand resource provisioning and teardown
+- No need for environment variables to be set up
+
+### In Scope for Scenario 2
+
+- **All [Scenario 1 activities](./0-scenario-1.spec.md#workflow) are included**
+- All five languages: .NET, Java, JavaScript, Python, Go
+- **Both [Net-New SDKs](#net-new-sdk) and existing SDKs**
+- TypeSpec-based generation for the Health Deidentification service
+- Both **data plane and management plane** coverage
+- Both **Agent Mode and CLI Mode** validation
+- All models available through GitHub Copilot:
+  - **Required for testing**: Claude Sonnet 4, Claude Sonnet 4.5, GPT-4o, GPT-5
+- Automated environment remediation (install/upgrade tooling)
+- **[TypeSpec Customizations](#typespec-customizations)** to shape generated surfaces
+- **[Code Customizations](#code-customizations)** to add handwritten code layers
+- Creation of **[Live Tests](#live-test)**, and new **[Test Recordings](#test-recording)** for [Net-New SDKs](#net-new-sdk)
+- Re-recording tests for existing SDKs
+- Automated handling of Azure resources for live tests
+
+### Out of Scope for Scenario 2
+
+- Outer-loop activities such as committing changes, creating PRs, release management, or package publishing
+- GA releases
+- Architectural review preparation
+  - **Note**: Even though architectural review preparation is out of scope for this scenario, instruction files should still provide guidance on how users can initiate the architectural review process outside of the agent workflow until it can be integrated into future scenarios
+  - **Note**: Similarly, package naming approval is required for [Net-New SDKs](#net-new-sdk), and instruction files should provide guidance on how to initiate the package naming approval process outside of the agent workflow
+- Breaking change reviews
+- Updating changelog/`README.md`/metadata content for data-plane libraries
+  - **Note**: For .NET, metadata update functionality (export-api script) already works for both data-plane and management-plane libraries and is implemented in Scenario 1
+- Error resolution assistance beyond environment setup remediation
+
+---
+
+## Workflow
```

Comment:
Did we consider "checkpoint by commit" during the full workflow?

Steps of azsdk_package_generate_code, azsdk_customized_code_update, azsdk_package_run_tests (and some other steps) would produce new code / modify existing code / add live tests / modify test asset files.

As a developer, it would be preferrable to do a commit after a step, save the state if next step went wrong, and/or make a clean env for next step.

We can consider this as a no-goal for now, and let developer handle it by themselves.